<a href="https://colab.research.google.com/github/akashgangadharan/My-learnings/blob/main/Parkinsons_disease.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Import Relevant Packages:
import pandas as pd
from sklearn import preprocessing
from xgboost import XGBClassifier
from sklearn.preprocessing import MinMaxScaler
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score


* The dataset for this project can be found from https://archive.ics.uci.edu/ml/machine-learning-databases/parkinsons/parkinsons.data  
* As you can see the data is in the form of link. So we will import the data directly from link.

In [ ]:
#Loading the dataset from URL
url= "https://archive.ics.uci.edu/ml/machine-learning-databases/parkinsons/parkinsons.data"
df=pd.read_csv(url)
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

* Now the data is imported. Lets start the instantiation process
* Lets make two Dataframes
1. Features (We have to delete **Status**(that is a label) and **name**(this is just an id; and have no significance in predicting the disease)
2. Labels (Here the label is **Status**; which essentially tells us whether the patient has Parkisons Disease or not)

In [ ]:
features=df.drop(['status','name'],axis=1)
label=df['status']

Lets count the value of each label

In [ ]:
label[label==1].shape[0], label[label==0].shape[0]

(147, 48)

* So there is 147 Positive and 48 Negative cases

We have to normalize the features dataset. For this we will use Min_Max scaler. It will scale down the dataset with specified range.  
The fit_transform() method fits to the data and transforms it.

In [ ]:
#Scaling the features between 1 and -1
scaler=MinMaxScaler((-1,1))
X=scaler.fit_transform(features)
y=label

Next, we will split the data in Training set and test set

In [ ]:
X_train,X_test,y_train,y_test=train_test_split(X,y,random_state=2, test_size=0.2)

In [ ]:
model=XGBClassifier()
model.fit(X_train,y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

In [ ]:
y_preds=model.predict(X_test)
accuracy_score(y_test,y_preds)

0.8205128205128205